File: src/main/java/model/equipment/weapon/Knife.java
package model.equipment.weapon;

import model.support.allEnums.DmgType;
import model.support.allEnums.WeaponType;

public class Knife extends Weapon {
    public Knife(double weight, int quality) {
        super(WeaponType.KNIFE, weight, quality);
        isOneHanded = true;
        setDamage();
    }

    private void setDamage() {
        damage.put(DmgType.CUT, quality * 0.50);
        damage.put(DmgType.SLASH, quality * 0.10);
        damage.put(DmgType.PIERCING, quality * 0.40);
    }
}

File: src/main/java/model/equipment/weapon/Knife.kt
package model.equipment.weapon

import model.support.enums.Skill
import model.support.enums.Skill.*
import model.support.enums.Specification.*
import model.support.enums.WeaponType.KNIFE

class Knife(weight: Double, quality: Int) : OneHandedWeapon(KNIFE, KNIFE_SKILL, ONE_HANDED_WEAPON_SKILL, weight, quality, true) {
    init {
        damage = mapOf(
                BLUNT to 0.0,
                CUT to quality * 0.50,
                SLASH to quality * 0.10,
                PIERCING to quality * 0.40,
                CRUSH to 0.0)
    }
}
